src/Integration.Vsix.UnitTests/CFamily/MsvcDriverTest.cs

```diff
@@ -760,5 +760,80 @@ public void AbsolutePathConversion()
             MsvcDriver.Absolute("root", "\r\t\nx:\\subdir1/subDir2\\xxx.foo")
                     .Should().Be("x:\\subdir1/subDir2\\xxx.foo");
         }
+
+        [TestMethod]
+        [DataRow("19.00.24210", "", false)] // Proper two Phase lookup not supported
+        [DataRow("19.11.25507", "", false)] // Two phase lookup is supported
+        [DataRow("19.00.24210", "/permissive-", false)]
+        [DataRow("19.11.25507", "/permissive-", true)]
+        [DataRow("19.11.25507", "/std:c++latest", false)]
+        [DataRow("19.11.25507", "/Zc:twoPhase-", false)]
+        [DataRow("19.11.25507", "/permissive- /Zc:twoPhase-", false)]
+        [DataRow("19.28.29333", "", false)] // No-permissive is default for latest
+        [DataRow("19.28.29333", "/permissive-", true)]
+        [DataRow("19.28.29333", "/std:c++latest", true)]
+        [DataRow("19.28.29333", "/permissive", false)]
+        [DataRow("19.28.29333", "/Zc:twoPhase-", false)]
+        [DataRow("19.28.29333", "/permissive- /Zc:twoPhase-", false)]
+        [DataRow("19.28.29333", "/std:c++latest /Zc:twoPhase-", false)]
+        [DataRow("19.28.29333", "/std:c++latest /permissive /Zc:twoPhase-", false)]
+        public void NoDelayedTemplateParsing(string compilerVersion, string cmd, bool expected)
+        {
+            var cmdList = new List<string>{ "cl.exe" };
+            cmdList.AddRange(cmd.Split(' '));
+            cmdList.Add("test.cpp");
+
+            var request = MsvcDriver.ToRequest(new[]
+            {
+                new CFamilyHelper.Capture
+                {
+                    Executable = "",
+                    StdOut = "",
+                    CompilerVersion = compilerVersion,
+                    X64 = true
+                },
+                new CFamilyHelper.Capture
+                {
+                    Executable = "",
+                    Cwd = "basePath",
+                    Env = new List<string>(),
+                    Cmd = cmdList
+                }
+            });
+
+            HasFlag(request, Request.NoDelayedTemplateParsing).Should().Be(expected);
+        }
+
+        [TestMethod]
+        [DataRow("/std:c++17", true)] // enabled by default
+        [DataRow("/std:c++latest", true)] // enabled by default
+        [DataRow("/std:c++17 /Zc:alignedNew", true)]
+        [DataRow("/std:c++latest /Zc:alignedNew", true)]
+        [DataRow("/std:c++17 /Zc:alignedNew-", false)]
+        [DataRow("/std:c++latest /Zc:alignedNew-", false)]
+        [DataRow("/std:c++14 /Zc:alignedNew", false)] // not supported in old languages
+        [DataRow("/Zc:alignedNew", false)] // not supported in old languages
+        public void AlignedNew(string cmd, bool expected)
```

Comment:
Ported from https://github.com/SonarSource/sonar-cpp/blob/3dc9a31ba6a8d66e2e113c2d64290ef0e23cf9ae/sonar-cfamily-plugin/src/test/java/com/sonar/cpp/analyzer/MsvcDriverTest.java#L325
and
https://github.com/SonarSource/sonar-cpp/blob/3dc9a31ba6a8d66e2e113c2d64290ef0e23cf9ae/sonar-cfamily-plugin/src/test/java/com/sonar/cpp/analyzer/MsvcDriverTest.java#L351